api/trace.proto

```diff
@@ -52,3 +53,12 @@ message ZipkinConfig {
   // /api/v1/spans, which is the default value.
   string collector_endpoint = 2 [(validate.rules).string.min_bytes = 1];
 }
+
+message DynamicConfig {
+  // Dynamic library implementing the `OpenTracing API
+  // <https://github.com/opentracing/opentracing-cpp>`_.
+  string library = 1 [(validate.rules).string.min_bytes = 1];
+
+  // File containing the configuration for the tracing library.
+  string config_file = 2 [(validate.rules).string.min_bytes = 1];
```

Comment:
Might this be more useful operationally if it was closer to `DataSource` (https://github.com/envoyproxy/data-plane-api/blob/master/api/sds.proto#L27), as it would avoid having to do filesystem management? Envoy could write files out to the filesystem if need be, i.e. if we can't load directly from memory into the trace driver.

Reply:
+1 we should allow inline config. Can we just use Struct here and use JSON, etc. for this? Or is that too restrictive. At minimum agreed we should use Datasource to allow inline configuration.

Reply:
Switched to use an inline configuration. (Looks like we'll be ok with having tracers use JSON for dynamic configurations.)